artemis-core-client/src/main/java/org/apache/activemq/artemis/core/client/impl/ClientConsumerImpl.java

```diff
@@ -710,10 +710,8 @@ public void clear(boolean waitForOnMessage) throws ActiveMQException {
       synchronized (this) {
          // Need to send credits for the messages in the buffer
 
-         Iterator<ClientMessageInternal> iter = buffer.iterator();
-
-         while (iter.hasNext()) {
-            try {
+         try (LinkedListIterator<ClientMessageInternal> iter = buffer.iterator()) {
```

Comment:
Funny! I had the exact same fix on my laptop ready to send a PR for :) 

if I merge your PR, it won't even show a difference